### PR TITLE
Add semantic query filters to Node SDK

### PR DESCRIPTION
**Pull request by MattTimms**
- Add semantic query filters to Node SDK
- Change Python SDK semantic query parameters for consensus across SDKs
- Added Python SDK test dependencies to pyproject.toml

### CODE REVIEW

File: sdks/python/src/redactive/search_client.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ async def query_chunks(
         access_token: str,
         semantic_query: str,
         count: int = 1,
-        query_filter: dict | None = None,
```
Comment (russellwmy) [inline sdks/python/src/redactive/search_client.py:36]: This may affect how the user was using it. We may need to notify  Python SDK users.

**Review by russellwmy — CHANGES_REQUESTED** [review]

File: sdks/python/src/redactive/search_client.py
Comment (IsisChameleon) [inline sdks/python/src/redactive/search_client.py:1]: QQ: No changes in the python tests for this addition of filters?

Comment (MattTimms) [reply sdks/python/src/redactive/search_client.py:1]: Filters was already implemented for the Python SDK including tests.

File: sdks/python/src/redactive/search_client.py (hunk 1 above)
Comment (MattTimms) [reply sdks/python/src/redactive/search_client.py:36]: Agreed, Rob has suggested a deprecation warning (see changes), but for next minor release I'd like to drop `query_filter` for `filters`

**Review by russellwmy — APPROVED** [review]
LGTM

**Review by kochie — APPROVED** [review]
LGTM

**Pull request merged**